.github/workflows/publish.yml

```diff
@@ -8,8 +8,50 @@ on:
       - v[0-9]+.[0-9]+.[0-9]+
 
 jobs:
+  prePublishPackageTest:
+    name: Prepublish package test
+    runs-on: ubuntu-latest
+    timeout-minutes: 30 # Default is 360
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Retrieve dependencies from cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: npm-v14-${{ runner.os }}-${{ github.ref }}-${{ hashFiles('package.json') }}
+
+      - name: Install Node.js and npm
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+          registry-url: https://registry.npmjs.org
+
+      - name: Install dependencies
+        if: steps.cacheNpm.outputs.cache-hit != 'true'
+        run: |
+          npm update --no-save
+          npm update --save-dev --no-save
+
+      - name: Build local package
+        run: npm pack
+
+      - name: Run tests against version packaged with npm pack
+        run: |
+          PACKAGE_VERSION=$(npm view . version)
+          mkdir "serverless-${PACKAGE_VERSION}"
+          tar zxf "serverless-${PACKAGE_VERSION}.tgz" -C "serverless-${PACKAGE_VERSION}"
+          cp -R test "serverless-${PACKAGE_VERSION}/package"
+          cp -R node_modules "serverless-${PACKAGE_VERSION}/package"
```

Comment:
I think we do not need to copy `node_modules`, they should be resolved just fine from parent folder, isn't it?

Reply:
I thought so as well but during local testing that wasn't the case - I also just double checked on other branch and it's failing: https://github.com/serverless/serverless/runs/1935331719?check_suite_focus=true

Reply:
I see, it's purely because `npm test` doesn't find mocha in its context.

Technically we can change test command to:

```
script -e -c "./node_modules/.bin/mocha serverless-${PACKAGE_VERSION}/package/test/unit/**/*.test.js"`
```

One maintainance issue with this, is that it's likely to break, if we decide to change `npm test` command (I'm pretty sure we will forget to update reference here).

Other cleaner approach could be to link `node_modules` instead of copying it.

What do you think?

Reply:
That's correct - I was thinking about the approach with just using `mocha` directly as you've shown, but I wasn't sold on not using standard `npm test` script from `package.json` for the reason you mentioned. I didn't think of linking - I've changed that and here's verification that it works correctly - https://github.com/serverless/serverless/runs/1935540414?check_suite_focus=true